### PR TITLE
Sburchfield33 johnny t06 mailchimp

### LINKED ISSUE #156: Mailchimp
# Description

Integrate a Mailchimp flow to allow admins to automate email sending.

## Specification

1. When user subscribes to the elist from the landing page, the application syncs the new email to Mailchimp. If the rate limit ends up being a problem, we can explore [crontask](https://vercel.com/templates/next.js/vercel-cron).
2. Write down all the specific steps on how to use Mailchimp.

### PR DESCRIPTION
**Pull request by sburchfield33**
# Description

Integrated the Mailchimp API into The Legacy Project. Injected code in the email POST route so email is not only uploaded to the database but also Mailchimp. 

## Issues
"Resolve #156"

## Test

You can subscribe on the public home page and then play around with test emails on the mailchimp website.

## Possible Downsides

On the free edition, Mailchimp is limited to 500 contacts and 1000 emails per month. So at 500 contacts you can send 2 emails per month. 

## Additional Documentations

Check out [https://docs.google.com/document/d/1-wO4pUNOSJt50RfE5KGLxuSrv3V_qeW328Ht-80mBsI/edit?usp=sharing](this) guide on sending letters with mailchimp.

Note that because there are account passwords in this doc I made it share only so be sure to share it with Arielle or any other admin. 

## Additional Tasks
Give me (Stephen) the names of the people that signed up to the elist so I can add them to the Mailchimp account. (because they were added before the mailchimp integration)

### CODE REVIEW

File: src/server/service/index.ts [hunk 1]
```diff
@@ -11,3 +12,10 @@ export const driveV3 = google.drive({
     scopes: ["https://www.googleapis.com/auth/drive"],
   }),
 });
+
+mailchimp.setConfig({
+  apiKey: env.MAILCHIMP_API_KEY,
+  server: "us22",
```
Comment (nickbar01234) [inline src/server/service/index.ts:18]: How did you determine `us22`?

Comment (johnny-t06) [reply src/server/service/index.ts:18]: I believe this is statically determined by mailchimp per account and "the last digits" of the API key.

Comment (nickbar01234) [reply src/server/service/index.ts:18]: Ah ok. Should we extract the server based of `MAILCHIMP_API_KEY`? Maybe in `server.schema` create a new key called `MAILCHIMP_SERVER`?

Comment (nickbar01234) [reply src/server/service/index.ts:18]: what's considered "the last digits"? Can we always determine the server prefix based on the API key?

Comment (johnny-t06) [reply src/server/service/index.ts:18]: "if the URL is https://us6.mailchimp.com/account/api/, then the data center subdomain is us6. It’s also appended to your API key in the form key-dc; if your API key is 0123456789abcdef0123456789abcde-us6, then the data center subdomain is us6." from https://mailchimp.com/developer/marketing/docs/fundamentals/#:~:text=The%20root%20url%20for%20the,mailchimp.com%2F3.0%2F%20.

Looks like extracting should work!

**Review by nickbar01234 — APPROVED** [review]
LGTM! Elegant solution!

**Pull request merged**